### ISSUE TITLE
Create an assertZeroAllocs test helper

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#4964]
```go
func assertZeroAllocs(t *testing.T, f func(), msgAndArgs ...interface{}) {
	t.Helper()
	assert.Equal(t, 0.0, testing.AllocsPerRun(5, f, msgAndArgs...)
}
```
I suggest doing it as a separate PR so that more people also review this refactoring (and merge this PR as it is).

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-go/pull/4957#discussion_r1499787721_

**Comment by MrAlias** [comment]
Blocked by https://github.com/open-telemetry/opentelemetry-go/pull/4957

**Comment by pellared** [comment]
Closing per https://github.com/open-telemetry/opentelemetry-go/pull/4965#pullrequestreview-1898693657
